### PR TITLE
React: add analytics geography view

### PR DESCRIPTION
**Pull request by const314**

### CODE REVIEW

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { Item } from 'devextreme-react/toolbar';
+import * as mapsData from 'devextreme/dist/js/vectormap-data/usa.js';
+import { RevenueByStatesCard } from './cards/RevenueByStatesCard';
+import { RevenueAnalysisByStatesCard } from './cards/RevenueAnalysisByStatesCard';
+import { RevenueSnapshotByStatesCard } from './cards/RevenueSnapshotByStates';
+
+const createMapCoords = (coords: string) => {
+  const resultCoords: number[] = [];
+
+  coords.split(', ').forEach((coord) => {
```
Comment (dxvladislavvolkov) [inline packages/react/src/pages/analytics-geography/analytics-geography.tsx:18]: Looks like we can use map method here

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          },
+          properties: {
+            text: item.stateName,
+            value: item.total,
+            tooltip: `<b>${item.stateName}</b>\n${item.total}K`,
+          },
+        })),
+      });
+    });
+    console.log({ salesByStateAndCity, salesByState });
```
[the remark this answers is not in the document]
Comment (const314) [reply packages/react/src/pages/analytics-geography/analytics-geography.tsx]: fixed

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx (hunk 1 above)
Comment (const314) [reply packages/react/src/pages/analytics-geography/analytics-geography.tsx]: fixed

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ANALYTICS_PERIODS[DEFAULT_ANALYTICS_PERIOD_KEY].period.split('/')
+  );
+  const [salesByStateAndCity, setSalesByStateAndCity] = useState([]);
+  const [salesByState, setSalesByState] = useState([]);
+  const [salesByStateMarkers, setSalesByStateMarkers] = useState({});
+
+  useEffect(() => {
+    getSalesByStateAndCity(...dateRange).then((data) => {
+      setSalesByStateAndCity(data);
+      const salesByStateResult = getSalesByState(data);
```
Comment (dxvladislavvolkov) [inline packages/react/src/pages/analytics-geography/analytics-geography.tsx:31]: Can we get rid of the variable and use the salesByState?

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      setSalesByState(salesByStateResult);
+      setSalesByStateMarkers({
```
Comment (dxvladislavvolkov) [inline packages/react/src/pages/analytics-geography/analytics-geography.tsx:33]: This object looks big. Maybe move this to some function to get the data?

Comment (const314) [reply packages/react/src/pages/analytics-geography/analytics-geography.tsx]: fixed

File: packages/react/src/pages/analytics-geography/analytics-geography.tsx (hunk 3 above)
Comment (const314) [reply packages/react/src/pages/analytics-geography/analytics-geography.tsx:48]: No. salesByState is a React state and cannot be mutated directly. We need to use the function result twice, so we need a temporary variable here.

**Review by MikeVitik — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**